src/Proxy.NET/Services/ProxyService.cs

```diff
@@ -0,0 +1,111 @@
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using Proxy.NET.Models;
+
+namespace Proxy.NET.Services;
+
+public class ProxyService(IHttpClientFactory httpClientFactory, IMetricService metricService, IConfiguration configuration) : IProxyService
+{
+    private const int HttpTimeoutSeconds = 60;
+    private bool EnableUnitTesting => configuration.GetValue<bool>("EnableUnitTesting");
+
+    /// <summary>
+    /// Sends an HTTP POST request with the specified JSON object to the specified request URL using the provided endpoint key.
+    /// </summary>
+    /// <param name="requestJson">The JSON object to send in the request body.</param>
+    /// <param name="requestUrl">The URL of the request.</param>
+    /// <param name="endpointKey">The endpoint key to use for authentication.</param>
+    /// <returns>A tuple containing the response content and the status code of the HTTP response.</returns>
+    public async Task<(string responseContent, int statusCode)> HttpPostAsync(
+        Uri requestUrl,
+        string endpointKey,
+        JsonDocument requestJsonDoc,
+        RequestContext requestContext
+    )
+    {
+        var httpClient = httpClientFactory.CreateClient();
```

Comment:
I think this is an `IDisposable` so we should be doing:

```suggestion
        using var httpClient = httpClientFactory.CreateClient();
```